### PR TITLE
add Jenkins Makefile

### PR DESCRIPTION
**Pull request by ChiamakaObitube**
Create a Jenkins template for setting up Continuous Integration for NodeJS, Go and Python application.

### CODE REVIEW

File: Jenkins/Go/Makefile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+test:
+		@echo 'Running vetting'
+		go vet .
+		@echo 'Running linting'
+		golint .
+		@echo 'Running test'
+		cd test && go test -v'
+
+build-image: #Build docker image .Here i have the Automart Docker file for build image.
+		${DOCKER} build -t ${DOCKER_HUB_REPO}/${APP_NAME}:$(RELEASE_TAG) .
```
Comment (haroldsphinx) [inline Jenkins/Go/Makefile:29]: Why don't you set /path/to/Dockerfile as env variables like you did for ${DOCKER_HUB_REPO}

and also shouldn't this be` DOCKER = docker ` path/to/docker ?

File: Jenkins/Go/Makefile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+push-image: #Push the Build image to DockerHUB. Make sure to login dockerHUB using  'docker login' ,before you push.
+		${DOCKER} push ${DOCKER_HUB_REPO}/${APP_NAME}:$(RELEASE_TAG)
+
+deploy:
+		@echo 'Deploy the app'
+		${DOCKER} run --name ${APP_NAME} -d -p 5000:5000 ${DOCKER_HUB_REPO}/${APP_NAME}:$(RELEASE_TAG)
```
Comment (haroldsphinx) [inline Jenkins/Go/Makefile:36]: Maybe you want to set the ports as env vars also, instead of hardcoding them

File: Jenkins/Node/Makefile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+test:
+	npm test
+
+build-image: #Build docker image .Here i have the Automart Docker file for build image.
+	${DOCKER} build -t ${DOCKER_HUB_REPO}/${APP_NAME}:$(RELEASE_TAG) .
+
+push-image: #Push the Build image to DockerHUB. Make sure to login dockerHUB using  'docker login' ,before you push.
+	${DOCKER} push ${DOCKER_HUB_REPO}/${APP_NAME}:$(RELEASE_TAG)
+
+deploy:
```
Comment (haroldsphinx) [inline Jenkins/Node/Makefile:24]: remember to update the port it here also

File: Jenkins/Go/Makefile (hunk 1 above)
Comment (ChiamakaObitube) [reply Jenkins/Go/Makefile:29]: Docker is the alias for thedocker hub credential on Jenkins

**Pull request merged**